pkg/cmd/infra/router/template.go

```diff
@@ -565,3 +566,88 @@ func (o *TemplateRouterOptions) blueprintRoutes(routeclient *routeclientset.Clie
 
 	return blueprints, nil
 }
+
+// makeTLSConfig checks whether metrics TLS is configured and
+// if so returns a tls.Config whose certificate is automatically
+// reloaded every 5 minutes or in response to file changes.
+func makeTLSConfig() (*tls.Config, error) {
+	certFile := env("ROUTER_METRICS_TLS_CERT_FILE", "")
+	if len(certFile) == 0 {
+		return nil, nil
+	}
+	keyFile := env("ROUTER_METRICS_TLS_KEY_FILE", "")
+
+	// Make sure we have something useful to start with
+	certificate, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		return nil, err
+	}
+
+	// Safely reload the certificate as requested.
+	var lock sync.Mutex
+	reload := make(chan bool) // value will be false for an unscheduled reload
+	go func() {
+		for {
+			select {
+			case scheduled := <-reload:
+				latest, err := tls.LoadX509KeyPair(certFile, keyFile)
+				if err != nil {
+					// TODO: add a prometheus metric? Fail a health check? This should surface somehow...
+					log.Error(err, "failed to reload certificate", "cert", certFile, "key", keyFile)
```

Comment:
The router has a lot of certificates, so using a specific phrase, such as "metrics certificate", here and below would reduce risk of confusion.